theano/tensor/elemwise.py

```diff
@@ -667,7 +668,7 @@ def grad(self, inputs, ograds):
         # TODO: make sure that zeros are clearly identifiable
         # to the gradient.grad method when the outputs have
         # some integer and some floating point outputs
-        if False in [str(out.type.dtype).find('int') == -1
+        if False in [str(out.type.dtype) in theano.tensor.continuous_dtypes
```

Comment:
This would be more clear:

    if any([out.type.dtype not in theano.tensor.continuous_dtypes for out in outs]):

otherwise a test would be great.